File: cc_prob.R
install.packages("readr")
library(readr)
cc<-read.csv(file.choose())
colnames(cc)<-c("y","x")
#####single variet analysis#####
attach(cc)
summary(x)
summary(y)
#barplot
barplot(x)
barplot(y)
#histogram
hist(x)
hist(y)
#box plot
boxplot(x,horizontal = T)
boxplot(y,horizontal = T)
#qqplot
qqnorm(x)
qqline(x)
qqnorm(log(y))
qqline(log(y))
shapiro.test(log(y))
shapiro.test(y)
######

install.packages("lattice")
library(lattice)
dotplot(x)
dotplot(y)
##correlation
cor(x,y)
####


#####multivariet analysis######
plot(x=y,y=x,main="calories vs weight",xlab = "calories",ylab = "weight",
     col="green",pch=20,col.lab="green",col.main="green")

model<-lm(y~x,data = cc)
summary(model)



shapiro.test(model$residuals)
err<-sqrt(mean(model$residuals^2))
err
#err is 103.25
#r-sqr value=0.89 and adjusted r sq value=0.88
#cor=0.94
#hence its best fit model

File: cc_prob.py
import pandas as pd 
import numpy as np
cc=pd.read_csv("C:/Users/USER/Desktop/cc.csv")
cc.columns="y","x"
import matplotlib.pylab as plt
####one variet analysis#####
plt.hist(cc.x)
plt.hist(cc.y)
plt.boxplot(cc.x)
plt.boxplot(cc.y)
cc.describe()
####bi variet analysis######
import scipy
from scipy import stats
stats.cor(cc.x,cc.y)

np.corrcoef(cc.x,cc.y)
plt.scatter(cc.x,cc.y)
import statsmodels.formula.api as smf
model=smf.ols("y~x",data=cc).fit()
model.summary()
pred=model.predict(pd.DataFrame(cc['x']))
pred
err=pred-cc.y
err_sq=err*err
err_mean=np.mean(err_sq)
err_sqrt=np.sqrt(err_mean)
err_sqrt
